src/utils/multipart.rs

```diff
@@ -46,3 +47,11 @@ pub fn read_multipart_sources(
             .and_then(|data| Ok(serde_json::from_slice(&data)?)),
     )
 }
+
+pub fn read_multipart_request_data(
```

Comment:
This + read_multipart_sources can be one generic fn

Reply:
Yes, that would seem sane but it appears I'm not smart enough to understand serde's lifetimes.  I'm just living with this for now instead of wasting loads of time.  I'd appreciate any hints though.